src/uwtools/scheduler.py

```diff
@@ -2,17 +2,66 @@
 Job Scheduling
 """
 
-from abc import abstractmethod
-import logging
 import collections
-from typing import Dict
+import logging
+from typing import Any, Dict, List
 
 logging.basicConfig(
     level=logging.DEBUG,
     format="%(asctime)s %(name)-12s %(levelname)-8s %(message)s",
     handlers=[logging.StreamHandler()],
 )
 
+NONEISH = [None, "", " ", "None", "none", False]
+IGNORED_ATTRIBS = ["scheduler"]
+
+
+class RequiredAttribs:  # pylint: disable=too-few-public-methods
+    """key for required attributes"""
+
+    ACCOUNT = "account"
+    QUEUE = "queue"
+    WALLTIME = "walltime"
+    NODES = "nodes"
+    TASKS_PER_NODE = "tasks_per_node"
+
+
+class OptionalAttribs:  # pylint: disable=too-few-public-methods
+    """key for optional attributes"""
+
+    SHELL = "shell"
+    JOB_NAME = "job_name"
+    STDOUT = "stdout"
+    STDERR = "stderr"
+    JOIN = "join"
+    PARTITION = "partition"
+    THREADS = "threads"
+    MEMORY = "memory"
+    DEBUG = "debug"
+    EXCLUSIVE = "exclusive"
+    CPUS = "cpus"
```

Comment:
What are `cpus`?

Reply:
leftover from trying to fill in the gaps mapping these.